src/worker/plugins/teardown.ts

```diff
@@ -0,0 +1,26 @@
+import { processError } from '../../shared/error'
+import { PluginsServer } from '../../types'
+
+export async function teardownPlugins(server: PluginsServer): Promise<void> {
+    const { pluginConfigs } = server
+
+    const teardownPromises = []
+    for (const [id, pluginConfig] of pluginConfigs) {
+        if (pluginConfig.vm) {
+            const teardownPlugin = await pluginConfig.vm.getTeardownPlugin()
+            if (teardownPlugin) {
+                teardownPromises.push(
+                    (async () => {
+                        try {
+                            await teardownPlugin()
+                        } catch (error) {
+                            await processError(server, pluginConfig, error)
```

Comment:
Hmm... I would keep it in. Otherwise we might lose valuable debugging data, especially at this early stage. The stacktrace should also show which function crashed, so it should be easy to ignore. If there's a lot of noise, we'll adjust.

Reply:
Sounds fair